LoveMachine.Core/ButtplugController.cs

```diff
@@ -195,6 +225,23 @@ protected void DoVibrate(float intensity, int girlIndex, Bone bone)
             client.VibrateCmd(intensity, girlIndex, bone);
         }
 
+        protected internal IEnumerator DoRotate(int girlIndex, Bone bone, bool clockwise, float strokeTimeSecs)
+        {
+            if (strokeTimeSecs == 0f)
+            {
+                client.RotateCmd(0, clockwise, girlIndex, bone);
+                yield break;
+            }
+
+            float downStrokeTimeSecs = strokeTimeSecs / 2f;
+
+            float downSpeed = Mathf.Lerp(0.3f, 1f, 0.4f / strokeTimeSecs) * CoreConfig.RotationSpeedRatio.Value;
+            float upSpeed = downSpeed * 0.8f;
+            client.RotateCmd(downSpeed, clockwise, girlIndex, bone);
+            yield return new WaitForSecondsRealtime(downStrokeTimeSecs);
+            client.RotateCmd(upSpeed, !clockwise, girlIndex, bone);

```

Comment:
1. In your experience, is this behavior better than the previous one?
2. Have you tried what it's like when you don't change direction here but keep everything else the same (only remove the "!")?

Reply:
> In your experience, is this behavior better than the previous one?
> Have you tried what it's like when you don't change direction here but keep everything else the same (only remove the "!")?

I tried both but TBH it's sameish for me.

I implemented both of them as you comment.
https://github.com/Sauceke/LoveMachine/pull/97#issuecomment-1124184472

Reply:
Then perhaps this could benefit from a setting for whether to change direction here?
`clockwise ^ CoreConfig.RotateBackwardsOnDownStroke.Value` instead of `!clockwise`, perhaps?
Just a suggestion though, it's your call.

Reply:
After several try, I feel `!clockwise` at pull back is important.